### PR TITLE
[BI-12] global search

### PR DESCRIPTION
**Pull request by Lucian2**
Am facut sa se actualizeze si pagina de Home, unde vin toate blogurile, in functie de ceea ce se cauta.
image

Daca nu exista niciun rezultat dupa cautare, am pus sa se afiseze un mesaj de eroare.

image

Am folosit Vue Suggestion pentru global search.

image

### CODE REVIEW

File: src/api/blogService.js [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
-        sort: sort
-    };
-    try {
-        let { data } = await axios.get('/api/restify/blogs', { params });
-        return data;
-    } catch (error) {
-        return error;
-    }
+export async function getBlogs(searchTerm, { page, perPage, sort }) {
+  let params = {
```
Comment (PascuMihai) [inline src/api/blogService.js:4]:
```jsx
let params = {
    page,
    perPage,
    sort,
  };
```
you can use the shorhand

File: src/api/blogService.js [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+    page: page,
+    perPage: perPage,
+    sort: sort,
+  };
+  try {
+    let { data } = await axios.get(
+      '/api/restify/blogs?sort=-views&search=' + searchTerm,
```
Comment (PascuMihai) [inline src/api/blogService.js:11]: you have the sort query in the params object and also search can go there also

Comment (PascuMihai) [reply src/api/blogService.js:11]: And the seach endpoint is different {{api}}/api/blogs/advance-search?search=Prof

File: src/api/blogService.js [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
+    );
+    return data;
+  } catch (error) {
+    return error;
+  }
 }
 
-export async function getBlog(blogId) {
-    return await axios.get(`/api/restify/blogs/${blogId}`)
+export async function getBlogSearchResults(searchTerm) {
```
Comment (PascuMihai) [inline src/api/blogService.js:20]: if we have search here why do we use search in the above function?

File: src/components/Tiptap/TiptapMenuBar.vue [hunk 4]
```diff
@@ -1,108 +1,299 @@
 <template>
   <div class="flex flex-wrap border">
     <tiptap-menu-item
-        :action="() => editor.chain().focus().toggleBold().run()"
```
Comment (PascuMihai) [inline src/components/Tiptap/TiptapMenuBar.vue:4]: not sure why we have this changes here, this should focus on search not on tip tap

File: src/components/BaseNavbar.vue [hunk 5]
```diff
@@ -45,12 +45,30 @@
...
-              class="block pl-10 border border-gray-300 rounded-md leading-5 bg-white placeholder-gray-500 focus:outline-none focus:placeholder-gray-400 focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500"
-            />
+              v-model="search"
+              display-attribute="attributes.title"
+              value-attribute="id"
+              @suggestion-click="goToBlog"
+              :list="getSearchResult"
+              :max-suggestions="5"
+              :debounce="500"
+              :controls="{
```
Comment (PascuMihai) [inline src/components/BaseNavbar.vue:58]: let's add this object into a computed

**Review by PascuMihai — APPROVED** [review]

**Review by PascuMihai — CHANGES_REQUESTED** [review]
There are some errois in the console
image

**Review by PascuMihai — APPROVED** [review]

**Review by Ceciliaa — APPROVED** [review]

**Pull request merged**